### PR TITLE
Sleep time is calculated based on xdf's file timestamp.

### PR DESCRIPTION
**Pull request by csavur**
- code improvement
- multi-stream functionality added.

### CODE REVIEW

File: XdfStreamer/xdfstreamer.cpp [hunk 1]
```diff
@@ -62,7 +60,7 @@ void XdfStreamer::pushRandomData(QSharedPointer<lsl::stream_outlet> outlet_ptr,
             }
         }
 
-        while (((double)clock()) / CLOCKS_PER_SEC < starttime + t * dSamplingInterval);
+        std::this_thread::sleep_for(std::chrono::milliseconds(dSamplingInterval));
```
Comment (cboulay) [inline XdfStreamer/xdfstreamer.cpp:63]: Shouldn't this be removed now that line 104 is in?

Comment (cboulay) [reply XdfStreamer/xdfstreamer.cpp:63]: oh sorry, I was looking at a partial diff. This is fine.

Comment (cboulay) [inline XdfStreamer/xdfstreamer.cpp]: This one will still have the problem discussed previously (eventually fewer samples pushed than expected based on elapsed time).

Something like:
`std::this_thread::sleep_for(std::chrono::millseconds(1000* (starttime + t*dSamplingInterval - ((double)clock()/CLOCKS_PER_SEC))));`

Please check.

Comment (csavur) [reply XdfStreamer/xdfstreamer.cpp]: Yes, I see what you mean. I did not think about the time that taking the push the sample to the lsl. I will make changes accordingly.

**Comment by csavur** [comment]
@cboulay 
I made the changes for sleeping time and push repo. Can you review it?

**Comment by cboulay** [comment]
Many apologies for letting this slip. While I didn't give it a thorough test, the code looks good to me and I tested it on one file that worked well enough. Thank you for your contribution and your patience!

**Pull request merged**